src/web/views/UMM/Code/index.jsx

```diff
@@ -0,0 +1,263 @@
+import React, { Component } from 'react'
+import CodeMirror from 'react-codemirror'
+import _ from 'lodash'
+
+import classnames from 'classnames'
+
+const style = require('./style.scss')
+
+require('codemirror/lib/codemirror.css')
+require('codemirror/theme/zenburn.css')
+require('codemirror/mode/yaml/yaml')
+
+const LAST_LINE_REGEX = /^(\w.+):/
+const REFRESH_INTERVAL = 1000 
+const TIME_EDITING = 2000
+const ANIM_TIME = 300
+const WAIT_TIME = 1500
+
+export default class CodeView extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loading: true,
+      lastEditTime: null
+    }
+
+    this.codeMirror = null
+    this.timer = null
+
+    this.isEditing = this.isEditing.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: false
+    })
+
+    setTimeout(() => { 
+      this.refreshPositionAdjustments() 
+    }, WAIT_TIME)
+
+    this.timer = setInterval(() => {
+      const codeChanged = !this.isEditing() && this.props.code !== this.state.lastCode
+      const needToRefresh = this.props.refresh
+
+      if (codeChanged || needToRefresh) {
+        this.props.setLoading(true)
```

Comment:
Additionally high components are 100% free to listen or not on these events, so it should be :
```js
this.props.onLoading && this.props.onLoading()
```